presto-main/src/main/java/com/facebook/presto/operator/NativeExecutionOperator.java

```diff
@@ -0,0 +1,247 @@
+
+    private boolean finished;
+
+    public NativeExecutionOperator(
+            PlanNodeId sourceId,
+            OperatorContext operatorContext,
+            JsonCodec<TaskSource> taskSourceCodec,
+            JsonCodec<PlanFragment> planFragmentCodec,
+            JsonCodec<TableWriteInfo> tableWriteInfoCodec,
+            PlanFragment planFragment,
+            TableWriteInfo tableWriteInfo,
+            PagesSerde serde)
+    {
+        this.sourceId = requireNonNull(sourceId, "sourceId is null");
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.systemMemoryContext = operatorContext.localSystemMemoryContext();
+        this.taskSourceCodec = requireNonNull(taskSourceCodec, "taskSourceJsonCodec is null");
+        this.planFragmentCodec = requireNonNull(planFragmentCodec, "planFragmentCodec is null");
+        this.tableWriteInfoCodec = requireNonNull(tableWriteInfoCodec, "tableWriteInfoCodec is null");
+        this.planFragment = requireNonNull(planFragment, "planFragment is null");
+        this.tableWriteInfo = requireNonNull(tableWriteInfo, "tableWriteInfo is null");
+        this.taskSource = null;
+        this.finished = false;
+        this.serde = serde;
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return false;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    /*
+     * TODO: The getOutput method will NativeExecutionTask APIs to:
+     *  1. submit the plan to the external process
+     *  2. wait on the getResult method from NativeExecutionTask until it returns (to get the task result from external process)
+     *  3. call getTaskInfo method to get the TaskInfo and propagate it
+     *  4. Deserialize the serialized page to presto Page and return it back
+     */
+    @Override
+    public Page getOutput()
+    {
+        if (finished) {
+            return null;
+        }
+        finished = true;
+        return null;
+    }
+
+    @Override
+    public void finish()
+    {
+        finished = true;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finished;
+    }
+
+    @Override
+    public PlanNodeId getSourceId()
+    {
+        return sourceId;
+    }
+
+    @Override
+    public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit split)
+    {
+        requireNonNull(split, "split is null");
+        checkState(this.taskSource == null, "NativeEngine operator split already set");
+
+        if (finished) {
+            return Optional::empty;
+        }
+
+        this.taskSource = new TaskSource(split.getPlanNodeId(), ImmutableSet.of(split), true);
+
+        Object splitInfo = split.getSplit().getInfo();
+        Map<String, String> infoMap = split.getSplit().getInfoMap();
```

Comment:
why do we have both getInfo and getInfoMap calls to get the information?

Reply:
Ah okay explained below. It's weird that the getInfo() is deprecated but still of some use.. Please ignore these 2 comments then